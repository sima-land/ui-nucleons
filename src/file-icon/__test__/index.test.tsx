import { render } from '@testing-library/react';
import { FileIcon } from '..';

describe('FileIcon', () => {
  const types = ['doc', 'xls', 'pdf', 'jpg', 'xml'];

  for (const type of types) {
    it('should render known file type icons', () => {
      const { container } = render(<FileIcon type={type} />);

      expect(container).toMatchSnapshot();
    });
  }

  it('should render unknown file type icon', () => {
    const { container } = render(<FileIcon type='cpp' />);

    expect(container).toMatchSnapshot();
  });
});
