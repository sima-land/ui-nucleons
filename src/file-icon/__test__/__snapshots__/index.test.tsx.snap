// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileIcon should render known file type icons 1`] = `
<div>
  <svg
    height="32"
    viewBox="0 0 32 32"
    width="32"
  >
    <path
      class="main doc"
      d="M6 5C6 3.89543 6.89543 3 8 3H20L26 9V27C26 28.1046 25.1046 29 24 29H8C6.89543 29 6 28.1046 6 27V5Z"
      opacity="0.88"
    />
    <path
      d="M21 9L26 9L20 3L20 8C20 8.55229 20.4477 9 21 9Z"
      fill="#fff"
      opacity="0.32"
    />
    <path
      d="M25.1667 9L21 9L26 15L26 9C25.5833 9 25.6269 9 25.1667 9Z"
      fill="black"
      opacity="0.24"
    />
    <text
      fill="#fff"
      font-size="6.5"
      font-weight="700"
      text-anchor="middle"
      x="16"
      y="23"
    >
      DOC
    </text>
  </svg>
</div>
`;

exports[`FileIcon should render known file type icons 2`] = `
<div>
  <svg
    height="32"
    viewBox="0 0 32 32"
    width="32"
  >
    <path
      class="main xls"
      d="M6 5C6 3.89543 6.89543 3 8 3H20L26 9V27C26 28.1046 25.1046 29 24 29H8C6.89543 29 6 28.1046 6 27V5Z"
      opacity="0.88"
    />
    <path
      d="M21 9L26 9L20 3L20 8C20 8.55229 20.4477 9 21 9Z"
      fill="#fff"
      opacity="0.32"
    />
    <path
      d="M25.1667 9L21 9L26 15L26 9C25.5833 9 25.6269 9 25.1667 9Z"
      fill="black"
      opacity="0.24"
    />
    <text
      fill="#fff"
      font-size="6.5"
      font-weight="700"
      text-anchor="middle"
      x="16"
      y="23"
    >
      XLS
    </text>
  </svg>
</div>
`;

exports[`FileIcon should render known file type icons 3`] = `
<div>
  <svg
    height="32"
    viewBox="0 0 32 32"
    width="32"
  >
    <path
      class="main pdf"
      d="M6 5C6 3.89543 6.89543 3 8 3H20L26 9V27C26 28.1046 25.1046 29 24 29H8C6.89543 29 6 28.1046 6 27V5Z"
      opacity="0.88"
    />
    <path
      d="M21 9L26 9L20 3L20 8C20 8.55229 20.4477 9 21 9Z"
      fill="#fff"
      opacity="0.32"
    />
    <path
      d="M25.1667 9L21 9L26 15L26 9C25.5833 9 25.6269 9 25.1667 9Z"
      fill="black"
      opacity="0.24"
    />
    <text
      fill="#fff"
      font-size="6.5"
      font-weight="700"
      text-anchor="middle"
      x="16"
      y="23"
    >
      PDF
    </text>
  </svg>
</div>
`;

exports[`FileIcon should render known file type icons 4`] = `
<div>
  <svg
    height="32"
    viewBox="0 0 32 32"
    width="32"
  >
    <path
      class="main jpg"
      d="M6 5C6 3.89543 6.89543 3 8 3H20L26 9V27C26 28.1046 25.1046 29 24 29H8C6.89543 29 6 28.1046 6 27V5Z"
      opacity="0.88"
    />
    <path
      d="M21 9L26 9L20 3L20 8C20 8.55229 20.4477 9 21 9Z"
      fill="#fff"
      opacity="0.32"
    />
    <path
      d="M25.1667 9L21 9L26 15L26 9C25.5833 9 25.6269 9 25.1667 9Z"
      fill="black"
      opacity="0.24"
    />
    <text
      fill="#fff"
      font-size="6.5"
      font-weight="700"
      text-anchor="middle"
      x="16"
      y="23"
    >
      JPG
    </text>
  </svg>
</div>
`;

exports[`FileIcon should render known file type icons 5`] = `
<div>
  <svg
    height="32"
    viewBox="0 0 32 32"
    width="32"
  >
    <path
      class="main xml"
      d="M6 5C6 3.89543 6.89543 3 8 3H20L26 9V27C26 28.1046 25.1046 29 24 29H8C6.89543 29 6 28.1046 6 27V5Z"
      opacity="0.88"
    />
    <path
      d="M21 9L26 9L20 3L20 8C20 8.55229 20.4477 9 21 9Z"
      fill="#fff"
      opacity="0.32"
    />
    <path
      d="M25.1667 9L21 9L26 15L26 9C25.5833 9 25.6269 9 25.1667 9Z"
      fill="black"
      opacity="0.24"
    />
    <text
      fill="#fff"
      font-size="6.5"
      font-weight="700"
      text-anchor="middle"
      x="16"
      y="23"
    >
      XML
    </text>
  </svg>
</div>
`;

exports[`FileIcon should render unknown file type icon 1`] = `
<div>
  <svg
    height="32"
    viewBox="0 0 32 32"
    width="32"
  >
    <path
      class="main"
      d="M6 5C6 3.89543 6.89543 3 8 3H20L26 9V27C26 28.1046 25.1046 29 24 29H8C6.89543 29 6 28.1046 6 27V5Z"
      opacity="0.88"
    />
    <path
      d="M21 9L26 9L20 3L20 8C20 8.55229 20.4477 9 21 9Z"
      fill="#fff"
      opacity="0.32"
    />
    <path
      d="M25.1667 9L21 9L26 15L26 9C25.5833 9 25.6269 9 25.1667 9Z"
      fill="black"
      opacity="0.24"
    />
    <g
      opacity="0.8"
    >
      <path
        d="M9 22H23V24H9V22Z"
        fill="#fff"
      />
      <path
        d="M9 18H19V20H9V18Z"
        fill="#fff"
      />
    </g>
  </svg>
</div>
`;
